vortex-buffer/src/buffer_mut.rs

```diff
@@ -423,19 +423,22 @@ impl<T> Extend<T> for BufferMut<T> {
 
         let remaining = self.capacity() - self.len();
 
-        let dst: *mut T = self.bytes.spare_capacity_mut().as_mut_ptr().cast();
-        let mut consumed = 0;
-        while consumed < remaining {
+        let begin: *const T = self.bytes.spare_capacity_mut().as_mut_ptr().cast();
+        let mut dst: *mut T = begin.cast_mut();
+        let buffer_end: *const T = unsafe { dst.add(remaining) };
+        while dst.addr() < buffer_end.addr() {
             if let Some(item) = iterator.next() {
-                // SAFETY: We know we have enough capacity to write the item.
-                unsafe { dst.add(consumed).write(item) };
-                consumed += 1;
+                unsafe {
```

Comment:
I think a comment saying that this is dark magic that shouldn't be touched would be appropriate